syntax = "proto3";

option java_package = "com.bitbouncer.streaming.grpc";
option objc_class_prefix = "bitbouncer_streaming";

package bitbouncer.streaming;

service streamprovider {
  rpc Subscribe (SubscriptionRequest) returns (stream SubscriptionBundle) {}
  rpc GetSchema (GetSchemaRequest) returns (GetSchemaReply) {}
}

message SubscriptionRequest {
  string topic = 1;
  int32 partition = 2;
  int64 offset = 3;
  int32 max_poll_time = 4;
}

message SubscriptionBundle {
  repeated SubscriptionData data = 1;
  bool  eof = 2;
}

message SubscriptionData {
  int32 key_schema=1;
  int32 value_schema=2;
  bytes key = 3;
  bytes value = 4;
  int64 offset = 5;
  int64 timestamp = 6;
}

// The request message containing the schema id
message GetSchemaRequest {
  int32 schema_id = 1;
}

// json encoded schema
message GetSchemaReply {
  string schema = 1;
}



